test/transform/ReferenceCollection.test.js

```diff
@@ -0,0 +1,266 @@
+  <sup id="cite_ref-c" class="reference"><a id='a3' href="#cite_note-c">[7]</a></sup>
+  <sup id="cite_ref-d" class="reference"><a id='a4' href="#cite_note-d">[8]</a></sup>
+  <span id="cite_note-a">0 1 2</span>
+  <span id="cite_note-b">3 4 5</span>
+  <span id="cite_note-c">6 7 8</span>
+  <span id="cite_note-d">9 10 11
+    <sup id="cite_ref-d">tick</sup>
+    <a class="mw-cite-backlink">link</a>
+  </span>
+`
+
+const prevSiblingGetter = ReferenceCollection.test.prevSiblingGetter
+const nextSiblingGetter = ReferenceCollection.test.nextSiblingGetter
+
+describe('ReferenceCollection', () => {
+  beforeEach(() => {
+    document = domino.createDocument(referenceGroupHTML)
+  })
+
+  describe('.collectNearbyReferenceNodes()', () => {
+    it('collects expected reference nodes', () => {
+      const sourceNode = document.querySelector('#cite_ref-b')
+      const nearbyRefNodes =
+        ReferenceCollection.test.collectNearbyReferenceNodes(sourceNode)
+
+      assert.deepEqual(nearbyRefNodes.map(node => node.id), [
+        'cite_ref-a',
+        'cite_ref-b',
+        'cite_ref-c',
+        'cite_ref-d'
+      ])
+    })
+  })
+
+  describe('.collectNearbyReferences()', () => {
+    it('collects expected references group and selected index', () => {
+
+      const MOCK_RECT = { top: 0, left: 1, width: 2, height: 3 }
+
+      // Domino doesn't implement 'getBoundingClientRect' so
+      // backfill it for testing methods which call it.
+      domino.impl.Element.prototype.getBoundingClientRect = () => MOCK_RECT
+
+      const secondAnchor = document.querySelector('#a2')
+      const nearbyReferences = ReferenceCollection.collectNearbyReferences(document, secondAnchor)
+
+      assert.equal(nearbyReferences.selectedIndex, 1)
+      assert.deepEqual(nearbyReferences.referencesGroup, [
+        { id: 'cite_ref-a',
+          rect: MOCK_RECT,
+          text: '[4]',
+          html: '0 1 2' },
+        { id: 'cite_ref-b',
+          rect: MOCK_RECT,
+          text: '[6]',
+          html: '3 4 5' },
+        { id: 'cite_ref-c',
+          rect: MOCK_RECT,
+          text: '[7]',
+          html: '6 7 8' },
+        { id: 'cite_ref-d',
+          rect: MOCK_RECT,
+          text: '[8]',
+          html: '9 10 11' }
+
+      ])
+    })
+  })
+  describe('.isCitation()', () => {
+    const isCitation = ReferenceCollection.isCitation
+    it('identifies citations', () => {
+      assert.ok(isCitation('bla#cite_note') === true)
+      assert.ok(isCitation('#cite_note') === true)
+    })
+    it('rejects non-citations', () => {
+      assert.ok(isCitation('bla#nope_note') === false)
+      assert.ok(isCitation('') === false)
+      assert.ok(isCitation('#') === false)
+    })
+  })
+
+  describe('.adjacentNonWhitespaceNode()', () => {
+    const adjacentNonWhitespaceNode = ReferenceCollection.test.adjacentNonWhitespaceNode
+    beforeEach(() => {
+      document.documentElement.innerHTML = `
+        <stuff>
+          \t<b id=one>one</b>
```

Comment:
Sorry but why an escaped tab here?

Reply:
The test is for `adjacentNonWhitespaceNode()`, so iirc I separated the elements with newlines and leading space "whitespace", so I thought it would be good to throw tabs in there too.